api/core/v1/core.proto

```diff
@@ -39,6 +39,9 @@ message ClusterObjectRef {
 
     // name of the cluster in which the resource exists
     string cluster_name = 3;
+
+    // annotations of the resource being referenced
+    map<string, string> annotations = 4;
```

Comment:
This feels a little strange to me. From the reference perspective I think we should still just have a `cluster_name`. No matter how it's represented in kubernetes.

Reply:
Agreed. I am removing it upon reconsideration.

At first, I did this because I found case in GME where ClusterObjectRef was being passed into a function that used a ClusterResourceId input param. So to create parity we could add that field to the proto so a GetAnnotations method would be created. But to date, this is the only discrepancy I've found, so its an easy fix to that function and its use, instead of polluting the protos.

Reply:
Actually this is still very much an issue throughout GME.